src/mscorlib/shared/System/Guid.cs

```diff
@@ -1232,80 +1282,77 @@ unsafe private static int HexsToCharsHexOutput(char* guidChars, int a, int b)
             return 9;
         }
 
-        // IFormattable interface
-        // We currently ignore provider
-        public String ToString(String format, IFormatProvider provider)
+        // Returns whether the guid is successfully formatted as a span. 
+        public bool TryFormat(Span<char> destination, out int charsWritten, string format = null)
         {
             if (format == null || format.Length == 0)
                 format = "D";
 
-            string guidString;
-            int offset = 0;
+            // all acceptable format strings are of length 1
+            if (format.Length != 1) 
+                throw new FormatException(SR.Format_InvalidGuidFormatSpecification);
+
             bool dash = true;
             bool hex = false;
-
-            if (format.Length != 1)
-            {
-                // all acceptable format strings are of length 1
-                throw new FormatException(SR.Format_InvalidGuidFormatSpecification);
+            bool setBraces = false;
+            bool curly = false;
+            int guidSize;
+
+            switch (char.ToUpperInvariant(format[0]))
```

Comment:
Why the change to ToUpperInvariant?  Did this make things faster than just using, e.g.
```C#
case 'D':
case 'd':
    ...;
    break;

case 'N':
case 'n':
   ...
   break;
```
etc.?

Reply:
This change didn't really impact TryFormat, but it improved ToString somewhat (by about 3.8%). TryParseGuidWithNoStyle also uses ToUpperInvariant in a check (although not in a switch statement), so using it here would be consistent.

Reply:
This looks suspicious. ToUpperInvariant is relatively expensive operation. switch should be much faster.